### PR TITLE
Button V2

### PR DESCRIPTION
**Pull request by ChengYanJin**
**Component**: button

**Description**:
image

Including keyboard navigation: 
image

@Cuervino 
I didn't find a way to add radius for the outline...
Do we open to `box-shadow` or we're OK with the current implementation?

**Design**:
You will need to check out the branch to play with the interaction.

**Breaking Changes**:

- [] Breaking Changes
none

In order to use the new button you will need to import it from this path
` import {Button} from "@scality/core-ui/dist/next"`

I removed all the snapshot in this PR, since it's not useful for us anymore. 

---

Closes: #ISSUE_NUMBER

### CODE REVIEW

File: src/lib/components/buttonv2/Buttonv2.component.js [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+function Button({
+  variant,
+  disabled,
+  label,
+  icon,
+  onClick,
+  tooltip,
+  ...rest
+}: Props) {
+  if (!icon && !label)
```
Comment (JBWatenbergScality) [inline src/lib/components/buttonv2/Buttonv2.component.js]: nit: add missing curly braces here

File: src/lib/components/buttonv2/Buttonv2.component.js [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    console.warn(
+      'Please specify either icon or label prop for the button component.',
+    );
+
+  if (icon && !tooltip)
```
Comment (JBWatenbergScality) [inline src/lib/components/buttonv2/Buttonv2.component.js]: nit add missing curly braces here

File: src/lib/components/buttonv2/Buttonv2.component.js [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+import Tooltip from '../tooltip/Tooltip.component.js';
+import type { Props as TooltipProps } from '../tooltip/Tooltip.component.js';
+import { getTheme } from '../../utils';
+import * as defaultTheme from '../../style/theme';
+import { spacing } from '../../style/theme.js';
+
+export type Props = {
+  variant?: 'primary' | 'secondary' | 'danger' | 'outline',
+  disabled?: boolean,
+  onClick?: (any) => void,
```
Comment (JBWatenbergScality) [inline src/lib/components/buttonv2/Buttonv2.component.js]: onClick takes a MouseEvent as an attribute (https://flow.org/en/docs/react/events/ , SyntheticMouseEvent<HTMLButtonElement>)

File: src/lib/components/buttonv2/Buttonv2.component.js [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+import type { Node } from 'react';
+import styled, { css } from 'styled-components';
+import Tooltip from '../tooltip/Tooltip.component.js';
+import type { Props as TooltipProps } from '../tooltip/Tooltip.component.js';
+import { getTheme } from '../../utils';
+import * as defaultTheme from '../../style/theme';
+import { spacing } from '../../style/theme.js';
+
+export type Props = {
+  variant?: 'primary' | 'secondary' | 'danger' | 'outline',
```
Comment (JBWatenbergScality) [inline src/lib/components/buttonv2/Buttonv2.component.js:12]: Variant is mandatory in your implementation

**Review by JBWatenbergScality — COMMENTED** [review]
It may be missing the loading/pending state

**Review by Cuervino — APPROVED** [review]
I could not find any flaw, looks very nice, thanks!

File: src/lib/components/buttonv2/Buttonv2.component.js (hunk 3 above)
Comment (ChengYanJin) [reply src/lib/components/buttonv2/Buttonv2.component.js]: oh. Yes. I missed that!

File: src/lib/components/buttonv2/Buttonv2.component.js (hunk 4 above)
Comment (ChengYanJin) [reply src/lib/components/buttonv2/Buttonv2.component.js:12]: Good catch! 
After talking with @Cuervino, we realized we still need the `ghost button`, meaning the variant should be optional. So I removed the warning in 8e4f8d35

**Pull request merged**